core/src/main/java/apoc/periodic/Periodic.java

```diff
@@ -177,8 +175,9 @@ public Stream<JobInfo> repeat(@Name("name") String name, @Name("statement") Stri
     }
 
     private void validateQuery(String statement) {
-        Util.validateQuery(db, statement,
-                READ_ONLY, WRITE, READ_WRITE);
+        Util.validateQuery(db, statement, 
+                Set.of(Mode.WRITE, Mode.READ, Mode.DEFAULT),
+                QueryType.READ_ONLY, QueryType.WRITE, QueryType.READ_WRITE);
```

Comment:
Now we will fail with `Supported query types for the operation are [READ_ONLY, WRITE, READ_WRITE]` regardless if it is the mode or querytype that is violated, do we want to write it so we fail with different error messages instead?

Reply:
Yes  it makes sense, I changed the `validateQuery` method in order to distinguish the two cases